gammapy/datasets/map.py

```diff
@@ -2694,6 +2729,36 @@ def to_map_dataset(self, name=None):
             meta_table=self.meta_table,
         )
 
+    def _to_asimov_dataset(self, name=None, models=None):
+        """Create Asimov dataset"""
+        npred = self.npred()
+        data = np.nan_to_num(npred.data, copy=True, nan=0.0, posinf=0.0, neginf=0.0)
+        npred.data = data.astype("float")
+
+        npred_off = get_wstat_mu_bkg(
```

Comment:
I am a bit confused. The `npred_off` gives the predicted counts in the off region, whereas `get_wstat_mu_bkg` adjusts this by the acceptance to return the background counts in the on region, no?

Reply:
looking at the property definitions `npred_background=self.alpha.data * get_wstat_mu_bkg` and `npred_off = npred_background/alpha` so `npred_off = get_wstat_mu_bkg`

Reply:
But is it really what we should be doing?

If you want to compute the sensitivity of a given observation with observed `counts_off` I think you have to keep the specific realization.

If it is a priori, you probably need to do it from a background model, like fake, no?